Mathlib/CategoryTheory/Enriched/Basic.lean

```diff
@@ -329,12 +343,50 @@ def EnrichedFunctor.forget {C : Type u₁} {D : Type u₂} [EnrichedCategory W C
     dsimp
     apply_fun ForgetEnrichment.homTo W
     · simp only [Iso.cancel_iso_inv_left, Category.assoc, tensor_comp,
-        ForgetEnrichment.homTo_homOf, EnrichedFunctor.map_comp, forgetEnrichment_comp]
+        ForgetEnrichment.homTo_homOf, EnrichedFunctor.map_comp, ForgetEnrichment.homTo_comp]
       rfl
     · intro f g w; apply_fun ForgetEnrichment.homOf W at w; simpa using w
 
+/-- `EnrichedFunctor.forget` distributes over composition of enriched functors up to isomorphism. -/
+@[simps!]
+def forgetComp (F : EnrichedFunctor W C D) (G : EnrichedFunctor W D E) :
+    (F.comp W G).forget ≅ F.forget ⋙ G.forget :=
+  NatIso.ofComponents (fun _ => Iso.refl _) (fun f => by simp [comp, forget])
+
+variable (W) (C) in
+/-- `EnrichedFunctor.forget` maps the identity enriched functor to a functor isomorphic to
+`Functor.id`. -/
+@[simps!]
+def forgetId : (EnrichedFunctor.id W C).forget ≅ Functor.id _ :=
+  NatIso.ofComponents (fun _ => Iso.refl _) (fun f => by simp [forget])
+
+/-- Enriched functors form a category with the morphisms between functors `F` and `G` being the
+natural transformations `F.forget ⟶ G.forget`. -/
+@[simps]
+instance category : Category (EnrichedFunctor W C D) where
+  Hom F G := F.forget ⟶ G.forget
```

Comment:
I feel a bit uneasy about this. This is very prone to defeq abuse and might potentially cause confusion for lean when trying to recognize morphisms (`α : F.forget ⟶ G.forget` will be infered as `_ ⟶ _` in two different categories). The previous definition with `EnrichedNatTrans` was so that no confusion was possible. You already see this in `EnrichedCat.lean` where you have to define the whiskering where you have to put `.forget`s everywhere in the constructions and hyps. I think the hom-types here should be made one-field structures to prevent any form of abuse. This is slightly more verbose but in my experience this is also more robust. (Potentially, this could also help auto-params in the construction of the bicategory of enriched categories).

Reply:
True, how about I just actually redefine `EnrichedNatTrans` to be this wrapper then. We can than expose it as a special case of `GradedNatTrans` either in this PR or whenever we need it.